scripts/extract-hooks.ts

```diff
@@ -110,6 +121,21 @@ async function execute(files: Array<string>) {
 
 exports.execute = execute;
 
+async function checkForInvalidHookLocations(patterns: string | Array<string>, ignore: Array<string>) {
+  const allFiles = await globby(patterns, {
+    absolute: true,
+    ignore,
+  });
+
+  const processPromises = [];
+  for (const absolutePath of allFiles)
+    processPromises.push(Promise.resolve().then(() => parseFile(absolutePath).then(file => processFile(file, {allowHooks: false}))));
+
+  await Promise.all(processPromises);
+}
+
+exports.checkForInvalidHookLocations = checkForInvalidHookLocations;
```

Comment:
While it was indeed an issue, I'm not sure adding a runtime check on all TS/TSX files is a good idea. If really needed, I feel like adding a lint rule preventing the use of `Hooks` as interface name (possibly via `@typescript-eslint/naming-convention`, which seems to allow regexps) would be enough.